modules/apps/portal-remote/portal-remote-json-web-service-web/src/main/java/com/liferay/portal/remote/json/web/service/web/internal/servlet/JSONWebServiceServlet.java

```diff
@@ -70,9 +71,16 @@ public void service(
 			 !path.equals(StringPool.SLASH)) ||
 			(httpServletRequest.getParameter("discover") != null)) {
 
-			LocaleThreadLocal.setThemeDisplayLocale(
-				_portal.getLocale(
-					httpServletRequest, httpServletResponse, true));
+			if (httpServletRequest.getParameter("ddm.locale") != null) {
+				LocaleThreadLocal.setThemeDisplayLocale(
+					LocaleUtil.fromLanguageId(
+						httpServletRequest.getParameter("ddm.locale")));
+			}
```

Comment:
@p-albuquerque please change it to:

```
String ddmDataProviderLanguageId =
	httpServletRequest.getParameter("ddmDataProviderLanguageId");

if (Validator.isNotNull(ddmDataProviderLanguageId)) {
	LocaleThreadLocal.setThemeDisplayLocale(
		LocaleUtil.fromLanguageId(ddmDataProviderLanguageId));
}
else
```

Reply:
I think it's better to use `Validator.isNotNull` instead of `!= null`

Reply:
Question: 
How can I know which one is better to check nullable: Validator, Objects, "!=", ....?
I know just that Validator method also check about empty(string) and zero(long) values

Reply:
In this case, `Validator.isNotNull` is better because `ddmDataProviderLanguageId` it's a string and we don't need to execute the code if it's empty or null. Does it make sense?

Reply:
Got it. But I mean in general cases

Reply:
Unfortunately it will depend on each case.